### PR TITLE
Add builder for options

### PR DESCRIPTION
**Pull request by senyer**
@jovanywang

### CODE REVIEW

**Comment by jovany-wang** [comment]
Add an unit test plz, thanks.

**Comment by senyer** [comment]
is done

**Comment by wenweihu86** [comment]
@senyer @jovany-wang there are many old usages on module brpc-java-examples, you should update it together.

**Comment by senyer** [comment]
Get it------------------ 原始邮件 ------------------
发件人: "Wenwei Hu"<notifications@github.com>
发送时间: 2019年11月28日(星期四) 上午9:37
收件人: "baidu/brpc-java"<brpc-java@noreply.github.com>;
抄送: "senyer"<576085903@qq.com>;"Mention"<mention@noreply.github.com>;
主题: Re: [baidu/brpc-java] Add builder for options (#148)



@senyer @jovany-wang there are many old usages on module brpc-java-examples, you should update it together.
 
&mdash;

File: brpc-java-core/src/main/java/com/baidu/brpc/server/RpcServerOptions.java [hunk 1]
```diff
@@ -87,6 +86,9 @@ public void setProtocolType(Integer protocolType) {
     // share global thread pool between multi rpcServer
     private boolean globalThreadPoolSharing = false;
 
+    @Tolerate
+    public RpcServerOptions(){}
```
Comment (jovany-wang) [inline brpc-java-core/src/main/java/com/baidu/brpc/server/RpcServerOptions.java]:
```suggestion
    public RpcServerOptions() {
    }
```

File: brpc-java-core/src/main/java/com/baidu/brpc/client/RpcClientOptions.java [hunk 2]
```diff
@@ -82,6 +83,9 @@
     // share worker thread poll and event thread pool between multi RpcClients
     private boolean globalThreadPoolSharing = false;
 
+    @Tolerate
+    public RpcClientOptions(){}
```
Comment (jovany-wang) [inline brpc-java-core/src/main/java/com/baidu/brpc/client/RpcClientOptions.java]:
```suggestion
    public RpcClientOptions() {
    }
```

File: brpc-java-core/src/test/java/com/baidu/brpc/RpcOptionsTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import com.baidu.brpc.protocol.Options;
+import com.baidu.brpc.server.RpcServerOptions;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class RpcOptionsTest {
+
+  @Test
+  public void testRpcClientOptionsBuilderMode() {
+    RpcClientOptions client = RpcClientOptions.builder()
```
Comment (jovany-wang) [inline brpc-java-core/src/test/java/com/baidu/brpc/RpcOptionsTest.java]:
```suggestion
    RpcClientOptions clientOptions = RpcClientOptions.builder()
```

File: brpc-java-core/src/test/java/com/baidu/brpc/RpcOptionsTest.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    Assert.assertEquals("UTF-8", client.getEncoding());
+    Assert.assertEquals(Options.ProtocolType.PROTOCOL_BAIDU_STD_VALUE, client.getProtocolType());
+    Assert.assertEquals(1000, client.getHealthyCheckIntervalMillis());
+    Assert.assertEquals(true, client.isReuseAddr());
+    Assert.assertEquals("client", client.getClientName());
+  }
+
+  @Test
+  public void testRpcServerOptionsBuilderMode() {
+    RpcServerOptions server = RpcServerOptions.builder()
```
Comment (jovany-wang) [inline brpc-java-core/src/test/java/com/baidu/brpc/RpcOptionsTest.java]:
```suggestion
    RpcServerOptions serverOptions = RpcServerOptions.builder()
```

**Comment by jovany-wang** [comment]
@senyer  Any progress on this PR?

**Pull request closed without merging**